### PR TITLE
Feature/test and documentation updates

### PR DESCRIPTION
**Pull request by jddeal**
This covered three things:
1. fleshed out README
2. reorganized tests to be in test suites
3. Added a lot of inline docs to the code for intellisense + our sanity in the future

The test suite thing seemed nice for bundling related things. Didn't change anything functionally though. One thing I didn't realize when I started doing it is that the test suite blocks are NOT general purpose blocks where you can run anything. It's more like defining struct/class member variables. Anything created at the test suite level is shared between the test cases, so what I tried here was usually just creating the ECS once, then maintaining its state throughout the test cases. The only real benefit is it saved the usual setup of ```
ECS ecs;
Entity entity = ecs.CreateEntity();```.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,130 @@
-# BabsECS
+# Babs-ECS
 [![Build Status](https://travis-ci.org/babsnbabs/babs-ecs.svg?branch=master)](https://travis-ci.org/babsnbabs/babs-ecs)
 
-Entity Component System implemented in C++, ideally with decent error handling and acceptable performance.
+A fast, simple, cross platform Entity Component System written for C++14.
```
Comment (misnor) [inline README.md]: with `std::any` we're technicall C++17 . This is what I have at the top of the repository:

`C++17 Header-Only Fast, Simple, Entity-Component-System`

I don't prefer one or the other, just make sure the C++ version is correct and make them line up (you should have admin rights to update the repro description.)

File: README.md [hunk 2]
```diff
@@ -1,16 +1,130 @@
...
+* [EntityPlus](https://github.com/Yelnats321/EntityPlus)
+* [EntityX](https://github.com/alecthomas/entityx)
+* [anax](https://github.com/miguelmartin75/anax)
+* [Artemis](https://github.com/vinova/Artemis-Cpp)
+
+This library will seem very familiar at its core and usage. It exists because we wanted a specific set of capabilities as well as the speed, but wanted a manageable codebase to work with in the event that we want to improve it. Some design goals included the following:
+* simple, small codebase
+* cross platform (linux, windows, and osx)
+* well documented and tested
+* fastest in the west
```
Comment (misnor) [inline README.md]: Reasonable errors? I don't think we have accomplished that yet unless the custom exception counts.

File: README.md [hunk 3]
```diff
@@ -1,16 +1,130 @@
...
+Event systems work well with ECS for de-coupled communication between systems. Events are as easy as components to work with. These don't require registration, and you can subscribe and broadcast at any time through the event manager provided by the ECS instance.
+
+```c++
+#include "babs-ecs/Manager.hpp"
+
+ECS ecs;
+
+struct CollisionEvent
+{
+	Entity e1;
```
Comment (misnor) [inline README.md]: Tabbing looks messed up between e1 and e2 here, can you confirm its good?

File: src/Manager.hpp [hunk 4]
```diff
@@ -49,6 +51,7 @@ class ECS {
 		this->entityIndex = 0;
 	}
 
+    // CreateEntity will initialize and return a new entity with no components.
```
Comment (misnor) [inline src/Manager.hpp]: Unsure if its github review or not but this comment doesnt look tabbed in enough

File: src/manager_tests.cpp [hunk 5]
```diff
@@ -21,182 +21,157 @@ struct Health
...
 
-	//still has Identity
-	auto identity = ecs.GetComponent(e0, Identity());
-	REQUIRE(identity->name == "babs2");
+        Entity e = ecs.CreateEntity();
 
-	auto entitiesWithHealth = ecs.EntitiesWith(Health());
-
-	REQUIRE(entitiesWithHealth.size() == 1);
+        // Should throw because AI is not registered
```
Comment (misnor) [inline src/manager_tests.cpp]: I wrote this comment but it doesn't make any sense.

"Should throw"
"However, it doesn't"

Then it checks that it DOES throw so it must, maybe delete the comment on ln 174?

File: README.md (hunk 1 above)
Comment (jddeal) [reply README.md]: `A fast, simple, and cross platform C++17 header-only Entity Component System`

File: README.md (hunk 3 above)
Comment (jddeal) [reply README.md]: fixed

File: src/Manager.hpp (hunk 4 above)
Comment (jddeal) [reply src/Manager.hpp]: Fixed, along with another couple lines

File: src/manager_tests.cpp (hunk 5 above)
Comment (jddeal) [reply src/manager_tests.cpp]: Oh yeah, thats not relevant anymore, I remember that from us debugging something

**Pull request merged**